### PR TITLE
Configure golangci-lint for src-cli

### PR DESCRIPTION
**Pull request by eseliger**
This has long been overdue. Finally, some sweet little fixes. The best one was a missing error check in our caching layer. :) Also adds this to CI.

### CODE REVIEW

File: cmd/src/search_stream.go [hunk 1]
```diff
@@ -16,11 +16,7 @@ import (
 	"github.com/sourcegraph/src-cli/internal/streaming"
 )
 
-var labelRegexp *regexp.Regexp
-
-func init() {
-	labelRegexp, _ = regexp.Compile("(?:\\[)(.*?)(?:])")
-}
+var labelRegexp = regexp.MustCompile(`(?:\[)(.*?)(?:])`)
```
Comment (eseliger) [inline cmd/src/search_stream.go:19]: Is this correct?

File: cmd/src/search_stream.go [hunk 2]
```diff
@@ -434,21 +429,21 @@ func streamSearchHighlightDiffPreview(diffPreview string, highlights []highlight
...
 	return strings.Join(final, "\n")
 }
 
 func stripMarkdownMarkers(content string) string {
-	content = strings.TrimLeft(content, "```COMMIT_EDITMSG\n")
-	content = strings.TrimLeft(content, "```diff\n")
-	return strings.TrimRight(content, "\n```")
+	content = strings.TrimPrefix(content, "```COMMIT_EDITMSG\n")
+	content = strings.TrimPrefix(content, "```diff\n")
+	return strings.TrimSuffix(content, "\n```")
```
Comment (eseliger) [inline cmd/src/search_stream.go:446]: Does this seem like a bug?

File: internal/batches/executor/execution_cache.go [hunk 3]
```diff
@@ -244,6 +244,9 @@ func (c ExecutionDiskCache) GetStepResult(ctx context.Context, key CacheKeyer) (
 	}
 
 	found, err := c.readCacheFile(path, &result)
+	if err != nil {
+		return result, false, err
+	}
```
Comment (eseliger) [inline internal/batches/executor/execution_cache.go:249]: 🔔

File: cmd/src/testdata/streaming_search_want.txt [hunk 4]
```diff
@@ -11,7 +11,7 @@
 [38;5;239m([0m[38;5;237mhttp://127.0.0.1:55128[0m[38;5;239m)
 [0m[0m[38;5;68msourcegraph/sourcegraph-atom > Stephen Gutekanst : all: release v1.0.5[0m[38;5;2m (1 match)[0m
 [38;5;239m--------------------------------------------------------------------------------
-[0m[38;5;239m[0m  oo [38;5;0m[48;5;11mbar[0m
+[0m[38;5;239m[0m  foo[38;5;0m[48;5;11m ba[0mr
```
Comment (eseliger) [inline cmd/src/testdata/streaming_search_want.txt:14]: 👁️

File: cmd/src/search_stream.go (hunk 1 above)
Comment (LawnGnome) [reply cmd/src/search_stream.go:19]: I believe so, yes.

File: cmd/src/search_stream.go (hunk 2 above)
Comment (LawnGnome) [reply cmd/src/search_stream.go:446]: Yep. Good catch!

File: cmd/src/testdata/streaming_search_want.txt (hunk 4 above)
Comment (LawnGnome) [reply cmd/src/testdata/streaming_search_want.txt:14]: :grimacing:

File: internal/batches/executor/execution_cache.go (hunk 3 above)
Comment (LawnGnome) [reply internal/batches/executor/execution_cache.go:249]: :see_no_evil:

**Review by LawnGnome — APPROVED** [review]
LGTM. Some good catches in here!

File: cmd/src/testdata/streaming_search_want.txt (hunk 4 above)
Comment (eseliger) [reply cmd/src/testdata/streaming_search_want.txt:14]: oo ar

**Pull request merged**